src/test/java/com/uber/cadence/workflow/WorkflowTest.java

```diff
@@ -938,6 +938,34 @@ public void testSyncUntypedAndStackTrace() {
         workflowClient.newUntypedWorkflowStub(
             "TestWorkflow1::execute", newWorkflowOptionsBuilder(taskList).build());
     WorkflowExecution execution = workflowStub.start(taskList);
+    testUntypedAndStackTraceHelper(workflowStub, execution);
+  }
+
+  @Test
+  public void testUntypedAsyncStart() throws Exception {
+    startWorkerFor(TestSyncWorkflowImpl.class);
+    WorkflowStub workflowStub =
+        workflowClient.newUntypedWorkflowStub(
+            "TestWorkflow1::execute", newWorkflowOptionsBuilder(taskList).build());
+    CompletableFuture<WorkflowExecution> future = workflowStub.startAsync(taskList);
+    testUntypedAndStackTraceHelper(workflowStub, future.get());
+  }
+
+  @Test
+  public void testUntypedAsyncStartWithTimeout() throws Exception {
+    startWorkerFor(TestSyncWorkflowImpl.class);
+    WorkflowStub workflowStub =
+        workflowClient.newUntypedWorkflowStub(
+            "TestWorkflow1::execute", newWorkflowOptionsBuilder(taskList).build());
+    Long timeout = new Long(200);
+    CompletableFuture<WorkflowExecution> future =
+        workflowStub.startAsyncWithTimeout(timeout, taskList);
+    testUntypedAndStackTraceHelper(workflowStub, future.get());
+  }
+
```

Comment:
Can you add a negative test where start returns exception? For example try to start workflow twice and expect the second call to error out.

Reply:
Added. 
Also added another test to first start then getResult both in async manners.